gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
```

Comment:
I don't understand the terminology here. How does it separates applications?

Reply:
Each type of application separator separates on its own, the interface is easy, you get a process and return whether you can tell which kind of application it is, and return it.